### PR TITLE
Datasets list

### PR DESCRIPTION
**Pull request by Vital-jan**
demo version without backend

### CODE REVIEW

File: src/components/pages/DatasetsList.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    <tr className="intro-x w-full">
+      <td className="w-4/5">
+        <h2 className="text-xl font-medium whitespace-no-wrap">
+          {name}
+        </h2>
+        {description}
+      </td>
+      <td className="w-1/5 align-middle">
+        <div className="text-center">
+          <button type="button" className="button button--md w-full flex items-center lg:justify-around md:justify-center sm:justify-center mr-1 mb-2 bg-theme-1 text-white">
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:21]: Use custom Button.jsx

File: src/components/pages/DatasetsList.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      <div className="top-bar" />
+      <h2 className="intro-y text-lg font-medium mt-10">
+        Перелік наборів даних
+      </h2>
+      <div className="flex items-center justify-between intro-y col-span-12 flex-wrap sm:flex-no-wrap mt-2 mb-4">
+        <div className="text-gray-600">Показано: з 1 по 10 з 150 елементів
+        </div>
+        <div className="sm:w-auto mt-3 sm:mt-0 sm:ml-auto md:ml-0">
+          <div className="w-56 relative text-gray-700">
+            <input type="text" className="input w-56 box pr-10 placeholder-theme-13" placeholder=" Пошук..." />
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:71]: Maybe use custom TextInput?

File: src/components/pages/DatasetsList.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          ))}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+
+DatasetsList.propTypes = {
+  ...ReactRouterPropTypes,
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:94]: You dont need ReactRouterPropTypes here

File: src/components/pages/DatasetsList.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import ReactRouterPropTypes from 'utils/react-router-prop-types';
+import PropTypes from 'prop-types';
+import { Search, Eye } from 'react-feather';
+
+const DatasetItem = (props) => {
+  const {
+    name, description,
+  } = props;
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:9]: do it in one line as there are few props

File: src/components/pages/DatasetsList.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      <td className="w-4/5">
+        <h2 className="text-xl font-medium whitespace-no-wrap">
+          {name}
+        </h2>
+        {description}
+      </td>
+      <td className="w-1/5 align-middle">
+        <div className="text-center">
+          <button type="button" className="button button--md w-full flex items-center lg:justify-around md:justify-center sm:justify-center mr-1 mb-2 bg-theme-1 text-white">
+            <span className="md:hidden sm:hidden lg:inline-block">Переглянути</span><span><Eye /></span>
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:22]: Use data list button from midone templates

Comment (roman-tiukh) [reply src/components/pages/DatasetsList.jsx:22]: And create component for this button

File: src/components/pages/DatasetsList.jsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+//         </h2>
+//         <h2>
+//           dfgdf dfgdfg dgdfgdf
+//         </h2>
+//       </div>
+//       <div className="col-span-2 flex items-center justify-center">
+//         <span>asd</span>
+//       </div>
+//     </div>
+//   );
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:42]: why is this commented out code here?

File: public/css/app.css [hunk 7]
```diff
@@ -1,7 +1,6 @@
 @charset "UTF-8";
 
 /* cyrillic-ext */
-
```
Comment (roman-tiukh) [inline public/css/app.css:4]: check what you commit carefully

File: src/components/pages/DatasetsList.jsx (hunk 2 above)
Comment (roman-tiukh) [reply src/components/pages/DatasetsList.jsx:71]: Or you can create SearchInput.jsx?

File: src/components/pages/DatasetsList.jsx (hunk 1 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: +

File: src/components/pages/DatasetsList.jsx (hunk 2 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: Used custom TextInput. Try to create SearchInput.

File: src/components/pages/DatasetsList.jsx (hunk 4 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx:11]: fixed;

File: src/components/pages/DatasetsList.jsx (hunk 5 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: +

File: src/components/pages/DatasetsList.jsx (hunk 3 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx:86]: Don`t understand yet

File: src/components/pages/DatasetsList.jsx (hunk 6 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: done;

File: public/css/app.css (hunk 7 above)
Comment (Vital-jan) [reply public/css/app.css:4]: ++

File: src/components/form-components/SearchInput.jsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  if (size) {
+    classList.push(`input--${size}`);
+  }
+  classList.push('input border');
+  classList.push('w-full');
+
+  if (formik && formik.touched[name] && formik.errors[name]) {
+    classList.push('error');
+  }
```
Comment (roman-tiukh) [inline src/components/form-components/SearchInput.jsx:24]: I think search input dont need formik support

File: src/components/form-components/SearchInput.jsx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+  const endId = id || `id_${name}`;
+
+  return (
+    <div className={`${width} mb-3 flex items-center justify-left relative inline-block`}>
+      {label && (
+        <label htmlFor={endId}>{label}</label>
+      )}
```
Comment (roman-tiukh) [inline src/components/form-components/SearchInput.jsx:32]: Is SearchInput need to have label?

Comment (roman-tiukh) [reply src/components/form-components/SearchInput.jsx:32]: I would consider the option of adding the searchBox=true/false property to TextInput

File: src/components/pages/DatasetsList.jsx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+import PropTypes from 'prop-types';
+import { Eye } from 'react-feather';
+import SearchBox from 'components/form-components/SearchBox';
+
+const DatasetPagination = (props) => {
+  const {
+    first,
+    last,
+    total,
+  } = props;
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:12]: code style: write this in one line

File: src/components/pages/DatasetsList.jsx [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        <div className="flex justify-center items-center">
+          <a
+            className="flex items-center mr-3"
+            href="/"
+            onClick={
+            (event) => {
+              event.preventDefault();
+              console.log(id);
+            }
+            }
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:49]: code style:
```
onClick={(event) => {
   event.preventDefault();
   console.log(id);
}}
```

File: src/components/pages/DatasetsList.jsx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+      Показано: з {first} по {last} з {total} елементів
+    </div>
+  );
+};
+
+DatasetPagination.propTypes = {
+  first: PropTypes.string.isRequired,
+  last: PropTypes.string.isRequired,
+  total: PropTypes.string.isRequired,
+};
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:24]: first, last, total can be PropTypes.number

File: src/components/pages/DatasetsList.jsx (hunk 4 above)
Comment (roman-tiukh) [reply src/components/pages/DatasetsList.jsx:31]: push to repo, if fixed.

File: src/components/pages/DatasetsList.jsx [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+        </div>
+      </td>
+    </tr>
+  );
+};
+
+DatasetItem.propTypes = {
+  name: PropTypes.string.isRequired,
+  date: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:63]: id can be number.

File: src/components/pages/DatasetsList.jsx [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+
+  return (
+    <div className="content">
+      <h2 className="intro-y text-lg font-medium mt-10">
+        Перелік наборів даних
+      </h2>
+      <div
+        className="
+          flex items-center justify-between intro-y
+          col-span-12 flex-wrap sm:flex-no-wrap mt-2 mb-4"
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:87]: code style
```
className={
  'flex items-center justify-between intro-y ' + 
  'col-span-12 flex-wrap sm:flex-no-wrap mt-2 mb-4'
}
```

File: src/components/pages/DatasetsList.jsx [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+      <div className="intro-y col-span-12 overflow-auto lg:overflow-visible">
+        <table className="table table-report intro-y -mt-2 white-space w-full">
+          <thead>
+            <tr>
+              <th className="whitespace-no-wrap">Набір даних</th>
+              <th className="text-center whitespace-no-wrap">Дата оновлення</th>
+            </tr>
+          </thead>
+          <tbody>
+            {datasetArray.map((item, number) => (
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:101]: rename "number" to "i" or "index"

File: src/components/pages/DatasetsList.jsx [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+  name: PropTypes.string.isRequired,
+  date: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
+};
+
+const DatasetsList = () => {
+  const datasetArray = [
+    { name: 'Реєстр адміністративно-територіального устрою',
+      date: '12/05/2020',
+      id: '01' },
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:70]: code style
```
{
  name: 'Реєстр адміністративно-територіального устрою',
  date: '12/05/2020',
  id: '01',
},
```

File: src/components/pages/DatasetsList.jsx [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+              <th className="text-center whitespace-no-wrap">Дата оновлення</th>
+            </tr>
+          </thead>
+          <tbody>
+            {datasetArray.map((item, number) => (
+              <DatasetItem
+                name={item.name}
+                date={item.date}
+                id={item.id}
+                key={`datasetItem${number}`}
```
Comment (roman-tiukh) [inline src/components/pages/DatasetsList.jsx:106]: `datasetItem${number}` is wrong way, use `item.id` for key

Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: ++

File: src/components/pages/DatasetsList.jsx (hunk 16 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: ++

File: src/components/pages/DatasetsList.jsx (hunk 15 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: ++

File: src/components/pages/DatasetsList.jsx (hunk 14 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: ++

File: src/components/pages/DatasetsList.jsx (hunk 13 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: ++

File: src/components/pages/DatasetsList.jsx (hunk 12 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx:21]: ++

File: src/components/pages/DatasetsList.jsx (hunk 11 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: ++

File: src/components/pages/DatasetsList.jsx (hunk 10 above)
Comment (Vital-jan) [reply src/components/pages/DatasetsList.jsx]: ++

**Pull request closed without merging**